format/src/main/java/com/github/sadikovi/riff/Statistics.java

```
@@ -472,4 +477,107 @@ public String toString() {
+    }
+
+    @Override
+    protected void updateState(InternalRow row, int ordinal) {
+      // value is guaranteed to be non-null
+      boolean value = row.getBoolean(ordinal);
+      if (hasValues) {
+        // if value is true, set max to true, otherwise set min to false
+        if (value) {
+          max = value;
+        } else {
+          min = value;
+        }
+      } else {
+        // for this update we have situation when min = max = true or min = max = false
+        min = value;
+        max = value;
+        hasValues = true;
+      }
+    }
+
+    @Override
+    protected void writeState(OutputBuffer buf) throws IOException {
+      // write 3 most significant bits as state of statistics
+      int flags = hasValues ? 1 : 0;
+      // set flags anyway, we will only read them if hasValues is set
+      flags |= min ? 2 : 0;
+      flags |= max ? 4 : 0;
+      buf.writeByte(flags);
+    }
+
+    @Override
+    protected void readState(ByteBuffer buf) throws IOException {
+      byte flags = buf.get();
+      hasValues = (flags & 1) != 0;
+      // flags are not considered if hasValues is set to false
+      min = (flags & 2) != 0;
+      max = (flags & 4) != 0;
+    }
+
+    @Override
+    protected void merge(Statistics obj) {
+      BooleanStatistics that = (BooleanStatistics) obj;
+      if (this.hasValues) {
+        this.min = that.hasValues ? (that.min && this.min) : this.min;
+        this.max = that.hasValues ? (that.max || this.max) : this.max;
+      } else {
+        // at this point values we assign do not really matter, because in case that.hasValues is
+        // false, then we will never check min/max values, otherwise they will be set correctly
+        this.min = that.min;
+        this.max = that.max;
+      }
+      this.hasValues = this.hasValues || that.hasValues;
+      // update nulls
+      this.hasNulls = this.hasNulls || that.hasNulls;
+    }
+
+    @Override
+    public boolean getBoolean(int ordinal) {
```

We might want to throw exception here, if statistics are not set.